dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/ModelChecker.java

```diff
@@ -168,4 +175,99 @@ protected void saveFlaggedPairsOutput(Wmm wmm, WmmEncoder encoder, ProverEnviron
         }
     }
 
+    protected static class LoggingProver implements ProverEnvironment {
+
+        private final FormulaManager manager;
+        private final ProverEnvironment inner;
+        private final List<String> log = new ArrayList<>();
```

Comment:
There should be a way to disable the logging, cause we generate huge formulas and having a full copy of the formula sitting in some logs seems to be wasteful. Indeed, I would prefer to have this disabled by default.

Reply:
This was not intended for upstream.  I have removed it.